compliance/lib/common/common.rego

```diff
@@ -112,8 +112,12 @@ file_in_path(path, file_path) {
 }
 
 # splits key value string by first occurrence of =
-split_key_value(key_value_string) = [key, value] {
-	seperator_index := indexof(key_value_string, "=")
+split_key_value(key_value_string, delimiter) = [key, value] {
+    # Validate input format
+    pattern = sprintf("--\\S+%s\\S+", [delimiter])
+	regex.match(pattern, key_value_string)
+
+	seperator_index := indexof(key_value_string, delimiter)
```

Comment:
I have to disagree with you, regex is almost never straightforward. I had to go to some regex website in order to understand it.

> what would you do differently?

I'm sure you tried it, but why even change it to regex? shouldn't it work with just the `delimiter` param you added? something like this:

```rego
split_key_value(key_value_string, delimiter) = [key, value] {
    # Validate input format
    seperator_index := indexof(key_value_string, delimiter)
    seperator_index != -1
```

Reply:
But I don't want to check that the delimiter exists, I want to check that the text matches the pattern.
I tried to avoid Regex where it is not necessary, but here it is essential since we want to validate that the input stands to some pattern.